### DISCUSSION TITLE
why custom lua plugins don't show in kong manager?

### DISCUSSION THREAD

**Discussion by solo-xwz** [Kong/kong#11212] [Help] [answered]
there are only official plugins

**Comment by brentos** [comment]
Did you add your plugin to the "plugins" configuration?
Example:
```
plugins = bundled,my-custom-plugin
```

**Reply by solo-xwz** [reply]
yes, I did, and I use the 'lua_package_path' configuration to install my plugins.
btw, I can apply these plugins using admin api, which means they are valid.

**Comment by Leopoldthecoder** [comment]
Hi @solo-xwz, could you navigate to the plugin selection page in your browser, open the dev tools, find the data returned from the `/:workspace/kong` endpoint, and confirm if `plugins.available_on_server` includes the plugin you want to install?

**Reply by solo-xwz** [reply]
Hi, i checked it and i found my custom plugins are not in `plugins.available_on_server` or `plugins.disabled_on_server`  but in `configuration.loaded_plugins`

**Reply by Leopoldthecoder** [reply]
Thanks for confirming. We'll see what we can do.

**Reply by Leopoldthecoder** [reply, marked as answer]
Hi @solo-xwz, we could not reproduce this problem (a custom plugin residing in `configuration.loaded_plugins` but not `plugins.available_on_server`), so could you provide detailed reproduction steps and your Kong Gateway config file?

If you are using the enterprise edition, you can contact your customer manager and they will help you collect the information needed. If not, let's continue the discussion here.

**Comment by luozhouyang** [comment]
same issue. 
![图片](https://github.com/Kong/kong/assets/34032031/2c55212e-b0d2-4d5f-b622-fbb02a8f0b15)

**Reply by osaka911** [reply]
same issue, did you fixed that?

**Reply by Leopoldthecoder** [reply]
Hi @luozhouyang @osaka911, this is a known regression. The next Kong Gateway release will fix it. For now, as a workaround, you can manually change the URL in your browser to create a custom plugin. For example, `/plugins/plugin-foo/create?redirect=/plugins`, assuming the name of your custom plugin is `plugin-foo`.

**Reply by osaka911** [reply]
@Leopoldthecoder thanks for your reply. When  does the next Kong release?

**Reply by Leopoldthecoder** [reply]
@osaka911 We are right now in the process of building the next version (Kong Gateway 3.7.0). And if things go smoothly it'll be available by the end of the month.

**Reply by osaka911** [reply]
Thanks

**Reply by luozhouyang** [reply]
Kong Gateway 3.7.1 fixed this issue.

**Comment by nehaDumbo15** [comment]
this issue is still not resolved ? 
Still we are not able to see the custom plugin in the list of all available plugin with the kong image **kong/kong-gateway**

**Reply by Just-Night** [reply]
The issue is fixed, just update to version 3.7.1

![fghjk](https://github.com/user-attachments/assets/590a0af3-2148-43c8-970b-3cd05a1fb413)

**Reply by nehaDumbo15** [reply]
NO  @Just-Night 
this is my configuration file
version: '3.9'

services:
 kong-migrate:
    image: kong/kong-gateway:3.7.1.0
    container_name: kong_migrate
    environment:
      KONG_DATABASE: postgres
      KONG_PG_HOST: postgres
      KONG_PG_PORT: 5432
      KONG_PG_USER: kong
      KONG_PG_PASSWORD: kong
      KONG_PG_DATABASE: kong
    command: kong migrations bootstrap
    entrypoint: [ "/bin/sh", "-c", "kong migrations bootstrap && touch /tmp/done" ]
    networks:
      - kong-net  
        
 kong-control: 
  depends_on:
      - kong-migrate
  image: kong/kong-gateway:3.7.1.0
  container_name: kong_control
  environment:
    KONG_DATABASE: postgres
    KONG_PG_HOST: postgres
    KONG_PG_PORT: 5432
    KONG_PG_DATABASE: kong
    KONG_PG_USER: kong
    KONG_PG_PASSWORD: kong
    KONG_ROLE: control_plane
    KONG_CLUSTER_CERT: /neha/cluster.crt
    KONG_CLUSTER_CERT_KEY: /neha/cluster.key
    KONG_ADMIN_LISTEN: "0.0.0.0:8001, 0.0.0.0:8444 ssl"
    KONG_PROXY_LISTEN: "0.0.0.0:8000, 0.0.0.0:8443 ssl"
    KONG_LUA_PACKAGE_PATH: /usr/local/custom/?.lua;;
    KONG_PLUGINS: bundled,neha-custom-plugin
  ports:
    - "8001:8001"
    - "8444:8444"
    - "8002:8002"
  command:  "kong start"
  networks:
    - kong-net
  volumes:
    - ./cluster.crt:/neha/cluster.crt
    - ./cluster.key:/neha/cluster.key  
    - ./plugins:/usr/local/custom/kong/plugins:rw

  restart: on-failure
     
        
 kong-data:
    depends_on:
      - kong-control

    image: kong/kong-gateway:3.7.1.0
    networks:
      - kong-net
    environment:
      KONG_DATABASE: off
      KONG_ROLE: data_plane
      KONG_CLUSTER_CERT: /neha/cluster.crt
      KONG_CLUSTER_CERT_KEY: /neha/cluster.key
      KONG_CLUSTER_CONTROL_PLANE: "kong-control:8005"
      KONG_CLUSTER_TELEMETRY_ENDPOINT: "kong-control:8006"
      KONG_PROXY_LISTEN: "0.0.0.0:8000, 0.0.0.0:8443 ssl"
      KONG_LUA_PACKAGE_PATH: /usr/local/custom/?.lua;;
      KONG_PLUGINS: bundled,neha-custom-plugin
    ports:
      - "8000:8000"
      - "8443:8443"
      - "8005:8005"
      - "8006:8006"
    volumes:
      - ./cluster.crt:/neha/cluster.crt
      - ./cluster.key:/neha/cluster.key  
      - ./plugins:/usr/local/custom/kong/plugins:rw

    restart: on-failure
  
  
networks:
  kong-net:
    external: true     

and 
![image](https://github.com/user-attachments/assets/7d9d368c-16f0-4167-98c6-e803571407c8)

**Reply by Just-Night** [reply]
"I use  `kong:3.7.1-ubuntu`. I had no problem with that.
Maybe you have some errors when starting kong?
```yml
...
  kong-cp:  # Control Plane
    container_name: kong-cp
    build:
      context: .
      dockerfile: Dockerfile
    environment:
      ...
      KONG_PLUGINS: bundled,...,http-to-https-redirect
      ...
...
```

```Dockerfile
FROM kong:3.7.1-ubuntu

USER root


COPY plugins /usr/local/share/lua/5.1/kong/plugins/
COPY scripts /usr/local/bin/

RUN chmod +x /usr/local/bin/migrate.sh


CMD ["kong", "docker-start"]
```

**Reply by Just-Night** [reply]
I apologize @nehaDumbo15 for my inattention you are using a different image

**Reply by nehaDumbo15** [reply]
No pblm  @Just-Night 
with this image kong:3.7.1-ubuntu we can see the list of custom plugin but with the image kong/kong-gateway we can't this is the main issue.
There is no confirmation for this mentioned in their docs if its because of free version or not

**Reply by Just-Night** [reply]
@nehaDumbo15  Do you use/plan to use a commercial version of kong?
There may be differences between commercial and community versions.
Since I see in the picture that you are currently using the free version

**Reply by nehaDumbo15** [reply]
@Just-Night 
Yes I am using the free version but this is not mentioned in their docs if its with enterprise version we can see or not the list of custom plugin.

**Reply by Just-Night** [reply]
@nehaDumbo15 
In my opinion, the problem may be that you are using image `kong/kong-gateway:3.7.1.0`
Does kong see your plugin?
Can you add it to any of the endpoints?

**Reply by nehaDumbo15** [reply]
@Just-Night 
Yes with the API call I can enable it with any service but not able to see in the list of all plugin

**Reply by Just-Night** [reply]
@nehaDumbo15 
I'm running the version `kong/kong-gateway:3.7.1.2`.
The problem is specifically with the `kong-gateway` version, most likely they haven't fixed it on it yet. 
@Leopoldthecoder can you help with this issue?
Also, if you don't plan to buy the enterprice version of kong in the future, I would recommend using `kong:3.7.1` or higher.
Of course, it has its limitations, but it all depends on what you plan to implement

**Reply by nehaDumbo15** [reply]
@Just-Night 
yes the issue with only kong-gateway image. 
Thanks for the help.

**Reply by robertoneto-senior** [reply]
Same problem here in `3.4.1.1` and `3.9.1.1`.

It is on `plugins.enabled_in_cluster`:

![image](https://github.com/user-attachments/assets/619d4ccc-b685-476c-a7f8-891c3f663c8f)

And also listed on `configuration.plugins`:

![image](https://github.com/user-attachments/assets/35030784-a5c6-4544-b499-49d68c26bd00)

And also on `configuration.loaded_plugins`:

![image](https://github.com/user-attachments/assets/a3d49ad7-4f14-4884-acf8-a0d4eab61c6d)

But not in `plugins.available_on_server`:

![image](https://github.com/user-attachments/assets/51464d5d-1191-494a-95b8-303138cf31bb)

**Comment by J1a-wei** [comment]
Hi @robertoneto-senior, Have u resovled this problem?

**Comment by robertoneto-senior** [comment]
Hi. @J1a-wei , I ~~solved~~ it migrating from the image `kong/kong-gateway:3.4.3.18` to `kong:3.4.2`.

This problem continues to occours. I'll open an issue.

**Comment by robertoneto-senior** [comment]
Issue created:  #14567
